### PR TITLE
Feat/shared volume

### PR DESCRIPTION
**Pull request by mirestrepo**
* Adding option to create an NFS volume to hold shared data. At the moment it requires to add changes to the helm chart as well (could be more modular #15, but I have terraform/jhub fatigue  )
* Add a related test  - this test is disabled on CI see #16 and test can be more modular but but I have terraform/jhub fatigue
* Fix a bug where poxy token was getting exposed in logs

For security reasons (self-hosted) tests are only triggered manually

### CODE REVIEW

**Comment by mirestrepo** [comment]
Okay, I'm ready for review! Basic test is passing - https://github.com/BrownUniversity/terraform-gcp-jupyterhub/actions/runs/239987950

If you have requests for changes (that require real work) and it's okay to postpone them, let's make issues. I need a vacation!

File: .kitchen.yml [hunk 1]
```diff
@@ -17,4 +17,10 @@
...
           command_timeout: 12000
-          parallelism: 1
\ No newline at end of file
+          parallelism: 1
+      # - name: sample-jhub-nfs
+      #   driver:
+      #     name: terraform
+      #     root_module_directory: examples/sample-jhub-nfs
+      #     command_timeout: 12000
+      #     parallelism: 1
```
Comment (broarr) [inline .kitchen.yml:26]: Do you need this commented out stuff?

File: modules/helm-jhub/variables.tf [hunk 2]
```diff
@@ -96,6 +96,37 @@ variable "auth_secretkeyvaluemap" {
...
+}
+
+variable "gcp_zone" {
+  type        = string
+  description = "The GCP zone to deploy the runner into."
+  default     = "us-east1-b"
+}
+
+variable "project_id" {
+
```
Comment (broarr) [inline modules/helm-jhub/variables.tf:127]: Should this have a type and description?

File: modules/shared-nfs/variables.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+variable "namespace" {
+  type        = string
+  description = "Name of kubernetes namespace to create the related resources"
+  default     = "jhub"
+}
+
+variable "annotations" {
+  type        = map
```
Comment (broarr) [inline modules/shared-nfs/variables.tf:32]: Any type needed here?

**Review by broarr — APPROVED** [review]
LGTM!

File: .kitchen.yml (hunk 1 above)
Comment (mirestrepo) [reply .kitchen.yml:26]: I don't :) but I wanted as a reminder that it's not enabled..  I can remove it though - I have an issue for re-enabling nfs testing

**Pull request merged**